php/api/endpoint/class-report.php

```diff
@@ -0,0 +1,140 @@
+	 *
+	 * @var string Base path.
+	 */
+	private $rest_base = 'report';
+
+	const POST_PATTERN     = '(?P<post_id>\d*)';
+	const CHECKSUM_PATTERN = '(?P<checksum>[a-fA-F\d]{64})';
+	const STANDARD_PATTERN = '(?P<standard>[\w-]+)';
+
+
+	/**
+	 * Register new routes.
+	 *
+	 * @action rest_api_init
+	 */
+	public function register_routes() {
+
+		register_rest_route( $this->namespace, '/' . $this->rest_base, array(
+			'methods'  => 'GET',
+			'callback' => array( $this, 'report_response' ),
+		) );
+
+		register_rest_route( $this->namespace, '/' . $this->rest_base . '/' . static::CHECKSUM_PATTERN . '/' . static::STANDARD_PATTERN, array(
+			'methods'  => 'GET',
+			'callback' => array( $this, 'report_response' ),
+		) );
+
+		register_rest_route( $this->namespace, '/' . $this->rest_base . '/' . static::POST_PATTERN . '/' . static::STANDARD_PATTERN, array(
+			'methods'  => 'GET',
+			'callback' => array( $this, 'report_response' ),
+		) );
+	}
+
+	/**
+	 * Prepare the response for a report requests.
+	 *
+	 * @param \WP_REST_Request $request The request object.
+	 *
+	 * @return mixed|\WP_REST_Response The response object.
+	 */
+	public function report_response( \WP_REST_Request $request ) {
+
+		// These reports are not for unauthenticated users.
+		if ( ! is_user_logged_in() ) {
+			return rest_ensure_response( $this->report_error( 'unauthenticated_call', __( 'unauthenticated report request', 'tide-api' ), 301 ) );
+		}
+
+		// @todo get the cloud storage source from configuration. For now this is AWS S3 only.
+		$object_source = 'aws_s3';
+
+		$checksum = $request->get_param( 'checksum' );
+		$post_id  = $request->get_param( 'post_id' );
+		$standard = $request->get_param( 'standard' );
+
+		// If we don't have a post_id, but a checksum, then hit the checksum endpoint.
+		if ( ! empty( $checksum ) ) {
+			$request  = new \WP_REST_Request( 'GET', '/' . $this->namespace . '/audit/' . $checksum );
+			$response = rest_do_request( $request );
+
+			if ( $response->is_error() ) {
+				// Don't be too verbose about the error.
+				return rest_ensure_response( $this->report_error( 'report_error', __( 'error occured in report api', 'tide-api' ), 301 ) );
+			}
+
+			// Set the post_id to get the meta for.
+			$post_id = $response->data['id'];
+		}
+
+		// Best effort to retrieve meta.
+		$meta = get_post_meta( $post_id, '_audit_' . $standard, true );
+
+		// Could not get the meta for the given standard.
+		if ( empty( $meta ) ) {
+			return rest_ensure_response( $this->report_error( 'report_standard_not_fount', __( 'could not retrieve report for standard', 'tide-api' ), 404 ) );
+		}
+
+		if ( empty( $this->plugin->components[ $object_source ] ) ) {
+			return rest_ensure_response( $this->report_error( 'report_source_error', __( 'could not retrieve report from source', 'tide-api' ), 404 ) );
+		}
+
+		// Get temporary signed url.
+		$meta = maybe_unserialize( $meta );
+		$url  = $this->plugin->components[ $object_source ]->get_url( $meta['full'] );
```

Comment:
Not sure we should be using `full` instead of the new data model. We need to update the Go services to send the correct payload, but we could merge this as-is and when the services are updated create another PR. Thoughts?

Reply:
There is no new data model yet.... `_audit_<standard>` is where we are pulling the object reference from... this is also the same place `summary` lives. It will require a bit more work to split them up.